### DISCUSSION TITLE
If WSL is enabled, it cannot be run

### DISCUSSION THREAD

**Discussion by shuzi48** [ppy/osu#30374] [Q&A] [answered]
### Type

Crash to desktop

### Bug description

After restarting the system, turn on WSL and turn on osu!. exe, the program cannot run.

### Version

2024.1009.1

### Logs

[Disable the logs after WSL starts OSU.zip](https://github.com/user-attachments/files/17436680/Disable.the.logs.after.WSL.starts.OSU.zip)

**Comment by shuzi48** [comment]
https://github.com/user-attachments/assets/8dff6085-85dc-455b-be9a-0872c03f0d02

**Comment by BlockHead007** [comment]
Can confirm there is no issues with lazer while WSL is running, tested on my system and it works just fine. More interesting is that lazer complains being running as administrator which should not happen in a normal scenario.

**Comment by bdach** [comment]
Nothing interesting in logs as far as I can see, game probably doesn't even launch properly for whatever reason to *start* logging anything. Please check windows event logs for anything related.

That said with the amount of custom plugins and rulesets you've got installed - some of them looking rather dodgy - I'm like 50% sure it's going to end up the fault of one of those.

**Comment by shuzi48** [comment]
> 据我所知，日志中没有什么有趣的事情，无论出于何种原因，游戏甚至可能无法正常启动以_开始_记录任何内容。请检查 Windows 事件日志以获取任何相关内容。
> 
> 也就是说，考虑到你安装的自定义插件和规则集的数量 - 其中一些看起来相当狡猾 - 我有 50% 的把握它最终会是其中一个的错。

I'm trying to find osu in the event viewer! Crash logs, but no osu!. exe

**Comment by shuzi48** [comment]
> 可以确认 WSL 运行时 lazer 没有问题，在我的系统上进行了测试，它运行良好。更有趣的是，lazer 抱怨以管理员身份运行，这在正常情况下不应该发生。
My system is streamlined, and it defaults to an administrator account

**Comment by bdach** [comment]
I'm sorry but "streamlined" system this, unsanctioned game plugins that... There's just no telling what's going on with your game at this point. Chances are high we cannot diagnose this for you and you'll have do some investigating yourself.

Downgrading to discussion because I see nothing actionable on our end. If the game didn't work on any computer with WSL I'd expect *a lot* more reports of that.

**Reply by HollisMeynell** [reply]
> I'm sorry but "streamlined" system this, unsanctioned game plugins that... There's just no telling what's going on with your game at this point. Chances are high we cannot diagnose this for you and you'll have do some investigating yourself.
> 
> Downgrading to discussion because I see nothing actionable on our end. If the game didn't work on any computer with WSL I'd expect _a lot_ more reports of that.

@bdach Will this issue be addressed in the future? #30783

**Comment by shuzi48** [comment]
在打开osu!后打开wsl,正常运行

**Comment by shuzi48** [comment, marked as answer]
The issue has been resolved, it is the WSL that is taking up some of the ports，Similar issue：https://github.com/ppy/osu/discussions/27651
